gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/BridgeServerAdvisor.java

```diff
@@ -66,7 +67,8 @@ protected Profile instantiateProfile(
 
   @Override
   public Set adviseProfileRemove() {
-    Set results = super.adviseProfileRemove();
+    Set results = new THashSet();
+    results.addAll(super.adviseProfileRemove());
```

Comment:
I think it will be better to change DistributionAdvisor.adviseFilter to always return a THashSet given that the empty case will be rare. It can be started with a smaller initial size say 4. This will avoid the unnecessary copying for the callers.